docs/src/main/asciidoc/reactive-postgres-client.adoc

```diff
@@ -0,0 +1,450 @@
+    }
+----
+
+And the endpoint to get all fruits from the backend:
+
+[source,java]
+.src/main/java/org/acme/vertx/FruitResource.java
+----
+@GET
+public CompletionStage<Response> get() {
+    return Fruit.findAll(client)
+            .thenApply(fruits -> fruits.isEmpty() ? Response.noContent() : Response.ok(fruits))
+            .thenApply(ResponseBuilder::build);
+}
+----
+
+Now start {project-name} in `dev` mode with:
+
+[source, shell]
+----
+mvn compile quarkus:dev
+----
+
+Eventually, open your browser and navigate to http://localhost:8080/fruits, you should see:
+
+[source, shell]
+----
+[{"id":3,"name":"Apple"},{"id":1,"name":"Orange"},{"id":2,"name":"Pear"}]
+----
+
+=== Prepared queries
+
+The Reactive Postgres Client can also prepare queries and take parameters that are replaced in the SQL statement at execution time:
+
+[source,java]
+----
+client.preparedQuery("SELECT name FROM fruits WHERE id = $1", Tuple.of(id))
+----
+
+TIP: The SQL string can refer to parameters by position, using $1, $2, ...etc.
+
+As the simple `query` method, `preparedQuery` eventually returns a `PgRowSet`.
+Equipped with this tooling, we are able to safely use an `id` provided by the user to get the details of a particular fruit:
+
+[source,java]
+.src/main/java/org/acme/vertx/Fruit.java
+----
+public static CompletionStage<Fruit> findById(PgPool client, Long id) {
+    return client.preparedQuery("SELECT id, name FROM fruits WHERE id = $1", Tuple.of(id)) // <1>
+            .thenApply(PgRowSet::iterator) // <2>
+            .thenApply(iterator -> iterator.hasNext() ? from(iterator.next()) : null); // <3>
+}
+----
+<1> Create a `Tuple` to hold the prepared query parameters.
+<2> Get an `Iterator` for the `PgRowSet` result.
+<3> Create a `Fruit` instance from the `Row` if an entity was found.
+
+And in the JAX-RS resource:
+
+[source,java]
+.src/main/java/org/acme/vertx/FruitResource.java
+----
+@GET
+@Path("{id}")
+public CompletionStage<Response> getSingle(@PathParam Long id) {
+    return Fruit.findById(client, id)
+            .thenApply(fruit -> fruit != null ? Response.ok(fruit) : Response.status(Status.NOT_FOUND)) // <1>
+            .thenApply(ResponseBuilder::build); // <2>
+}
+----
+<1> Prepare a JAX-RS response with  either the `Fruit` instance if found or the `404` status code.
+<2> Build and send the response.
+
+The same logic applies when saving a `Fruit`:
+
+[source,java]
+.src/main/java/org/acme/vertx/Fruit.java
+----
+public CompletionStage<Long> save(PgPool client) {
+    return client.preparedQuery("INSERT INTO fruits (name) VALUES ($1) RETURNING (id)", Tuple.of(name))
+            .thenApply(pgRowSet -> pgRowSet.iterator().next().getLong("id"));
+}
+----
+
+And in the web resource we handle the `POST` request:
+
+[source,java]
+.src/main/java/org/acme/vertx/FruitResource.java
+----
+@POST
+public CompletionStage<Response> create(Fruit fruit) {
+    return fruit.save(client)
+            .thenApply(id -> URI.create("/fruits/" + id))
```

Comment:
JAX-RS has an API for reverse-routing, but it's so bad that I'll let that slip ;)